snuba/query/data_source/simple.py

```diff
@@ -58,6 +59,7 @@ class Table(SimpleDataSource):
 
     table_name: str
     schema: PhysicalColumnSet
+    storage_key: StorageKey
```

Comment:
This allows us to get all the goodies on the storage. imo ideally we would just query from storages in the ClickhouseQuery (since all these things are properties of the storage). But I didn't want to change that in this PR as well